### ISSUE TITLE
Custom command `menuFromCommand` with filter gives unexpected result

### ISSUE THREAD

**Issue by andrelouw** [jesseduffield/lazygit#1577]
**Describe the bug**
Considering the following custom command:

```yaml
customCommands:
  - key : '<c-a>'
    description: 'Print commit message'
    command: "echo {{ index .PromptResponses 0 }}"
    subprocess: true
    context: 'commits'
    prompts:
      - type: 'menuFromCommand'
        title: 'Commits:'
        command: "git log --oneline"
        filter: '(?P<commit_id>[0-9a-zA-Z]*) *(?P<commit_msg>.*)'
        valueFormat: '{{ .commit_id }} - {{ .commit_msg }}'
        labelFormat: '{{ .commit_id | green | bold }} - {{ .commit_msg | yellow }}'
```

 The chosen commit will not be used as the result of the prompt. i.e. `{{ index .PromptResponses 0 }}` will not be the chosen option.
In this example the last commit in the list is always used. After trying a few different examples this seems to be the case everytime. The last option in the list is always used and not the chosen option.

I also tried the following alternative, using `.group_GROUP_ID` instead:

```yaml
customCommands:
  - key : '<c-a>'
    description: 'Print commit message'
    command: "echo {{ index .PromptResponses 0 }}"
    subprocess: true
    context: 'commits'
    prompts:
      - type: 'menuFromCommand'
        title: 'Commits:'
        command: "git log --oneline"
        filter: '(?P<commit_id>[0-9a-zA-Z]*) *(?P<commit_msg>.*)'
        valueFormat: '{{ .group_1 }} - {{ .group_2 }}'
        labelFormat: '{{ .group_1 | green | bold }} - {{ .group_2 | yellow }}'
```

**To Reproduce**
1. Add the snippet above in your `config.yml`
2. Open the `lazygit` repo 
3. Run the command with `Ctrl-a` from the `Commits` panel
4. Choose a commit from the list of commits
5. Note the result printed out

**Expected behavior**
The chosen commit should be printed out after the selection. (Not the `Initial Commit`)

**Screenshots**
| Step | Screenshot |
| --- | --- |
| Run command from commits panel | <img width="603" alt="screenshot 2021-11-13 at 08 14 15" src="https://user-images.githubusercontent.com/25477369/141610246-688ff359-a0ea-47cd-9612-824481d1f2ec.png"> |
| Select a commit from list | <img width="1032" alt="screenshot 2021-11-13 at 08 14 32" src="https://user-images.githubusercontent.com/25477369/141610247-a9b2cad4-7572-499e-ab2b-0ed99a225aee.png"> |
| Result printed out _(Not the chosen commit)_ | <img width="380" alt="screenshot 2021-11-13 at 08 28 45" src="https://user-images.githubusercontent.com/25477369/141610249-d377619d-9ba5-4212-88ad-32f57cafca33.png"> | 

**Desktop (please complete the following information):**
 - OS: `macOS` (`12.0.1` and earlier)
 - Lazygit Version: `0.31.3`

**Additional context**
I have been tracking this bug over the last few months on different OS versions and `lazygit` versions and the results are the same every time.

**Comment by Magellol** [comment]
I've been having the same issue with the following command. The last option in the generated list seems to always be the one that is chosen


```yaml
- type: menuFromCommand
   subprocess: true
   title: "Base branch?"
   command: "git branch"
   filter: ".*(.*)"
   valueFormat: "{{ .group_0 }}"
```

**Comment by jesseduffield** [comment]
This was fixed in v0.32. Note to anybody reading: whenever you have a bug in go where the last item in an array is always the one that's acted upon, it's because you failed to reassign a loop variable that you forgot to close over. Truly a bug in the language.

I'll close this off. Lemme know if you're still having problems!
